lib/btree.js

```diff
@@ -0,0 +1,240 @@
+    if (nextNode.length === minDegree) {
+      nextNode = growChild(currNode, 0);
+    }
+    currNode = nextNode;
+  }
+  return currNode.shift();
+};
+
+const extractLowerLimit = (node, minDegree) => {
+  let currNode = node;
+  while (!(isLeaf(currNode))) {
+    let nextNode = currNode[currNode.length - 1].child;
+    if (nextNode.length === minDegree) {
+      nextNode = growChild(currNode, currNode.length - 1);
+    }
+    currNode = nextNode;
+  }
+  const test = currNode.splice(currNode.length - 2, 1)[0];
+  return test;
+};
+
+const deleteElement = (node, elementIndex, minDegree) => {
+  const element = node[elementIndex];
+  const deletedData = element.data;
+  const leftChild = element.child;
+  const rightChild = node[elementIndex + 1].child;
+  if (leftChild.length > minDegree) {
+    const lowerLimit = extractLowerLimit(leftChild, minDegree);
+    element.value = lowerLimit.value;
+    element.data = lowerLimit.data;
+  } else if (rightChild.length > minDegree) {
+    const upperLimit = extractUpperLimit(rightChild, minDegree);
+    element.value = upperLimit.value;
+    element.data = upperLimit.data;
+  } else {
+    joinNodes(node, elementIndex, elementIndex + 1);
+    rightChild.splice(minDegree - 1, 1);
+  }
+  return deletedData;
+};
+
+class BTree {
+
+  constructor(degree = DEFAULT_DEGREE) {
+    this.root = [empty()];
+    this.minDegree = degree;
+  }
+
+  get(value) {
+    let currNode = this.root;
+    while (currNode) {
+      const [found, i] = binarySearch(currNode, value);
+      if (found) {
+        return currNode[i].data;
+      }
+      currNode = currNode[i].child;
+    }
+    return null;
+  }
+
+  insert(value, data, replace = false) {
+    const newElement = new Element(value, data);
+    if (this.root.length === 1) {
+      this.root.unshift(newElement);
+      return this;
+    }
+    if (this.root.length === this.minDegree * 2) {
+      this.root = [empty(this.root)];
+      splitNode(this.root, 0);
+    }
+    let currNode = this.root;
+    while (true) {
+      const [found, nextNodeIndex] = binarySearch(currNode, value);
+      if (!replace && found) {
+        throw new Error(`value ${value} already is in the tree`);
+      }
+      if (isLeaf(currNode)) {
+        currNode.splice(nextNodeIndex, 0, newElement);
+        return this;
+      }
+      let nextNode = currNode[nextNodeIndex].child;
+      if (nextNode.length === this.minDegree * 2) {
+        splitNode(currNode, nextNodeIndex);
+        const element = currNode[nextNodeIndex];
+        if (!replace && element.value === value) {
+          throw new Error(`value ${value} already is in the tree`);
+        }
+        if (element.value > value) {
+          nextNode = element.child;
+        }
+      }
+      currNode = nextNode;
+    }
+  }
+
+  iterator(start, finish) {
+    return inorderTraversal(start, finish, this.root);
+  }
+
+  remove(value) {
+    let currNode = this.root;
+    while (!isLeaf(currNode)) {
```

Comment:
Why don't you change this to `while (currNode)` (or !== null / !== undefined) and then you won't need the code on L232-L235. If that's because delete doesn't work for Leafs then I think adding a `!isLeaf` check in the last else branch on L152 might fix that.